### PR TITLE
feat: display failed role creation indication

### PR DESCRIPTION
**Pull request by junminahn**
In `Role Management` tab:
- highlight failed role creation items;
- show an option to retry.

### CODE REVIEW

File: app/page-partials/my-dashboard/ClientRoles.tsx [hunk 1]
```diff
@@ -332,10 +332,14 @@ const ClientRoles = ({ selectedRequest, alert }: Props) => {
         title="Create New Role"
         icon={null}
         onConfirm={async (contentRef: any) => {
-          await contentRef.current.submit();
-          await contentRef.current.reset();
-
-          reset();
+          const hasError = await contentRef.current.submit();
```
Comment (thegentlemanphysicist) [inline app/page-partials/my-dashboard/ClientRoles.tsx:335]: Not sure I'm following the logic here.  Does `contentRef.current.submit()` only resolve to something truthy if there's an error?  That seems strange

**Review by thegentlemanphysicist — COMMENTED** [review]
Just a bit confused by the has error block of code

File: app/page-partials/my-dashboard/ClientRoles.tsx (hunk 1 above)
Comment (junmin-bcgov) [reply app/page-partials/my-dashboard/ClientRoles.tsx:335]: I see, the `GenericModal` components passes a `ref` object into its children to make it available in `onConfirm` callback. In this case `CreateRoleContet` component is the one. `contentRef.current.submit` is a method in `CreateRoleContet` and always returns `boolean` type: https://github.com/bcgov/sso-requests/pull/544/files#diff-fc93ab1efab9b04e3b7bc1299f3026dc6602b3e57463f6031a57994ed353a93dR97

Comment (thegentlemanphysicist) [reply app/page-partials/my-dashboard/ClientRoles.tsx:335]: Oh that makes sense, thank you

**Review by thegentlemanphysicist — APPROVED** [review]

**Pull request merged**